ide/editor/demosrc/base/org/netbeans/editor/example/Editor.java

```diff
@@ -544,7 +544,7 @@ private String[] getOpenedFiles() {
             }
         }
 	
-        return (String []) opened.toArray(new String[opened.size()]);
+        return opened.toArray(new String[0]);
```

Comment:
Shouldn't this be String[opened.size()] that way things are pre-allocated?

Reply:
toArray() method have autosize operation on jvm side. You can check inside.

Reply:
> toArray() method have autosize operation on jvm side. You can check inside.

Sure it does.. But, the writer of the software expressly requested a pre-allocated String as an argument. This changes the intent of that work. Also, I don't really like to think about the internals. I like to think about what the author wanted done, which in this case is to expressly pre-allocate a String.

I would get rid of the cast and nothing more.

Reply:
@BradWalker this might be interesting: https://www.baeldung.com/java-collection-toarray-methods (the variant, that allocates a zero sized array wins)

Reply:
@BradWalker this old PR has also some info about that "false friend": 3166